From 51cd8d5f084013360b29fe497193288ebde2f4f9 Mon Sep 17 00:00:00 2001
From: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
Date: Wed, 29 Jul 2015 23:13:33 +0200
Subject: [PATCH 08/10] kodi-config.cmake: use CMAKE_FIND_ROOT_PATH to fix
 cross-compilation

When cross-compiling, the location at build time of the libraries is
not the same as the one at run-time. The CMAKE_FIND_ROOT_PATH variable
is here to handle this difference, so use it in kodi-config.cmake.

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
---
 project/cmake/KodiConfig.cmake.in | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/project/cmake/KodiConfig.cmake.in b/project/cmake/KodiConfig.cmake.in
index c02a680..9d6a566 100644
--- a/project/cmake/KodiConfig.cmake.in
+++ b/project/cmake/KodiConfig.cmake.in
@@ -7,10 +7,10 @@ if(NOT @APP_NAME_UC@_PREFIX)
   set(@APP_NAME_UC@_PREFIX @APP_PREFIX@)
 endif()
 if(NOT @APP_NAME_UC@_INCLUDE_DIR)
-  set(@APP_NAME_UC@_INCLUDE_DIR @APP_INCLUDE_DIR@)
+  set(@APP_NAME_UC@_INCLUDE_DIR ${CMAKE_FIND_ROOT_PATH}/@APP_INCLUDE_DIR@)
 endif()
 if(NOT @APP_NAME_UC@_LIB_DIR)
-  set(@APP_NAME_UC@_LIB_DIR @APP_LIB_DIR@)
+  set(@APP_NAME_UC@_LIB_DIR ${CMAKE_FIND_ROOT_PATH}/@APP_LIB_DIR@)
 endif()
 if(NOT @APP_NAME_UC@_DATA_DIR)
   set(@APP_NAME_UC@_DATA_DIR @APP_DATA_DIR@)
@@ -19,6 +19,7 @@ if(NOT WIN32)
   set(CMAKE_CXX_FLAGS "$ENV{CXXFLAGS} @CXX11_SWITCH@")
 endif()
 list(APPEND CMAKE_MODULE_PATH @APP_LIB_DIR@ @APP_DATA_DIR@/cmake)
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_FIND_ROOT_PATH}/@APP_LIB_DIR@ ${CMAKE_FIND_ROOT_PATH}/@APP_DATA_DIR@/cmake)
 
 string(REPLACE ";" " " ARCH_DEFINES "@ARCH_DEFINES@")
 add_definitions(${ARCH_DEFINES} -DBUILD_KODI_ADDON)
-- 
2.10.2

